crypto/signer.go

```diff
@@ -24,30 +24,25 @@ import (
 )
 
 var (
-	signerHashLookup = map[sigpb.DigitallySigned_HashAlgorithm]crypto.Hash{
-		sigpb.DigitallySigned_SHA256: crypto.SHA256,
-	}
-	reverseSignerHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
+	sigpbHashLookup = map[crypto.Hash]sigpb.DigitallySigned_HashAlgorithm{
 		crypto.SHA256: sigpb.DigitallySigned_SHA256,
 	}
 )
 
 // Signer is responsible for signing log-related data and producing the appropriate
 // application specific signature objects.
 type Signer struct {
-	hash         crypto.Hash
-	signer       crypto.Signer
-	sigAlgorithm sigpb.DigitallySigned_SignatureAlgorithm
+	hash   crypto.Hash
+	signer PrivateKeyManager
 }
 
 // NewSigner creates a new Signer wrapping up a hasher and a signer. For the moment
 // we only support SHA256 hashing and either ECDSA or RSA signing but this is not enforced
 // here.
-func NewSigner(sigAlgo sigpb.DigitallySigned_SignatureAlgorithm, signer crypto.Signer) *Signer {
+func NewSigner(signer PrivateKeyManager) *Signer {
```

Comment:
`PrivateKeyManager` exposes exactly what `Signer` needs (and no more), so it seems reasonable to use it here. The tests actually get a little simpler in general. I don't think it's onerous to ask callers to provide a `PrivateKeyManager`, but I could add a `NewFromSigner()` func to key_manager.go to make it extra easy to provide one if you have a `crypto.Signer` that isn't a `crypto.PrivateKey`.